yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnShuffleEncryptionSuite.scala

```diff
@@ -0,0 +1,244 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.yarn
+
+import java.io.File
+import java.io.FileInputStream
+import java.io.IOException
+import java.io.InputStream
+import java.io.OutputStream
+import java.util.ArrayList
+import java.util.LinkedList
+import java.util.List
+import java.util.Random
+import java.util.UUID
+import java.security.PrivilegedExceptionAction
+
+import scala.runtime.AbstractFunction1
+
+import com.google.common.collect.HashMultiset
+import com.google.common.io.ByteStreams
+import org.apache.hadoop.security.Credentials
+import org.apache.hadoop.security.UserGroupInformation
+import org.junit.Assert.assertEquals
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+import org.mockito.Answers.RETURNS_SMART_NULLS
+import org.mockito.Matchers.any
+import org.mockito.Matchers.anyInt
+import org.mockito.Mockito.doAnswer
+import org.mockito.Mockito.when
+import org.scalatest.{BeforeAndAfterAll, Matchers}
+
+import org.apache.spark._
+import org.apache.spark.crypto.CommonConfigurationKeys
+import org.apache.spark.crypto.CryptoConf
+import org.apache.spark.crypto.CryptoStreamUtils
+import org.apache.spark.deploy.SparkHadoopUtil
+import org.apache.spark.executor.ShuffleWriteMetrics
+import org.apache.spark.executor.TaskMetrics
+import org.apache.spark.io.CompressionCodec
+import org.apache.spark.memory.TestMemoryManager
+import org.apache.spark.memory.TaskMemoryManager
+import org.apache.spark.network.util.LimitedInputStream
+import org.apache.spark.serializer.KryoSerializer
+import org.apache.spark.serializer.SerializerInstance
+import org.apache.spark.shuffle.IndexShuffleBlockResolver
+import org.apache.spark.shuffle.sort.{SerializedShuffleHandle, UnsafeShuffleWriter}
+import org.apache.spark.storage.BlockId
+import org.apache.spark.storage.BlockManager
+import org.apache.spark.storage.DiskBlockManager
+import org.apache.spark.storage.DiskBlockObjectWriter
+import org.apache.spark.storage.TempShuffleBlockId
+import org.apache.spark.util.Utils
+
+class YarnShuffleEncryptionSuite extends SparkFunSuite with Matchers with BeforeAndAfterAll{
```

Comment:
This feels very removed from the code that's actually being changed. It feels like these tests should be in places like `DiskBlockObjectWriterSuite` and `BlockStoreShuffleReaderSuite`, since that's the code you're actually changing.